datadog_checks_dev/datadog_checks/dev/tooling/commands/test.py

```diff
@@ -151,6 +160,15 @@ def test(
             echo_debug(f"No envs found for: `{check}`")
             continue
 
+        ddtrace_check = ddtrace
+        if ddtrace and ON_WINDOWS and any('py2' in env for env in envs):
```

Comment:
Since all the environment are grouped in the tox command, the flag is disabled for the whole check if there is one windows-py2 env